nonos/main.py

```diff
@@ -680,503 +737,201 @@ def plot(
+                coordgrid[1],
+                coordgrid[2],
             )
-        )
-        h = 0.5 * l / np.sqrt(vr ** 2 + vt ** 2)
-
-        return sign * h * np.array([vr, vt / x])
-
-    def get_stream(
-        self,
-        vx,
-        vy,
-        x0,
-        y0,
-        nmax=1000000,
-        maxlength=4 * np.pi,
-        bidirectional=True,
-        reverse=False,
-    ):
-        """
-        Function for computing a streamline.
-        Parameters:
-        -----------
-
-        x0,y0: Floats.
-              Initial position for the stream
-        nmax: Integer.
-              Maxium number of iterations for the stream.
-        maxlength: Float
-                   Maxium allowed length for a stream
-        bidirectional=True
-                      If it's True, the stream will be forward and backward computed.
-        reverse=False
-                The sign of the stream. You can change it mannualy for a single stream,
-                but in practice, it's recommeneded to use this function without set reverse
-                and setting bidirectional = True.
-
-        Output:
-        -------
-
-        If bidirectional is False, the function returns a single array, containing the streamline:
-        The format is:
-
-                                          np.array([[x],[y]])
 
-        If bidirectional is True, the function returns a tuple of two arrays, each one with the same
-        format as bidirectional=False.
-        The format in this case is:
-
-                                (np.array([[x],[y]]),np.array([[x],[y]]))
+        if is_set(lic) and self.code != "fargo3d":
+            # TODO: careful, works for a cylindrical structure
+            # but may be wrong when other structures
+            # will be implemented (in particular the extent of interpolation)
+            if 1 in plane[:-1]:
+                if 3 in plane[:-1]:
+                    extent_i = (None, extent[1], extent[2], extent[3])
+                else:
+                    extent_i = (None, extent[1], None, None)
+            elif 3 in plane[:-1]:
+                extent_i = (None, None, extent[2], extent[3])
+            else:
+                extent_i = (None, None, None, None)
+
+            xi, yi, lici = LICstream(
+                self.init,
+                self.on,
+                plane=plane,
+                lines=lic,
+                func_proj=func_proj,
+                corotate=self.config["corotate"],
+                isPlanet=self.config["isPlanet"],
+                average=average,
+                xxmin=extent_i[0],
+                xxmax=extent_i[1],
+                yymin=extent_i[2],
+                yymax=extent_i[3],
+                dxx=licres,
+                dyy=licres,
+                kernel_length=30,
+                niter=2,
+            )
+            # print(f"xmin: {xi.min()}, xmax: {xi.max()}, ymin: {yi.min()}, ymax: {yi.max()}")
+            # print(f"extent: {extent}")
```

Comment:
I know this is commented code but it tells me that you don't know about "debug fstrings" (Python 3.8+ only)
```suggestion
            # print(f"{xi.min()=}, {xi.max()=}, {yi.min()=}, {yi.max()=}")
            # print(f"{extent=}")
```

Reply:
Indeed, I started to use it only recently